shared/shared.go

```diff
@@ -75,26 +75,38 @@ func MakeLabelFunc() func(hash LabelHash, labelID uint64, leftSiblings [][]byte)
 	}
 }
 
+type Member struct {
+	Challenge []byte `scale:"max=32"`
+}
+
 type Proof struct {
 	// The actual proof.
 	MerkleProof
 
 	// Members is the ordered list of miners challenges which are included
 	// in the proof (by using the list hash digest as the proof generation input (the statement)).
-	Members [][]byte
+	Members []Member `scale:"max=150"`
```

Comment:
Why 150? There could be much more members. A member is each node that submitted a challenge for the given round. In fact, there is no limit to how many members there can be in a round.

Reply:
That would be an issue, since we have to specify some maximum number of members. What's a realistic upper limit? 1 Mio?

Reply:
I'm not sure. 1M should be more than enough under normal conditions (unless the poet is spammed with challenges).